### PR TITLE
css_peace / first warden mute

### PR DESCRIPTION
**Pull request by P250**
- Added IEventService, see the class for more detail
- Added css_peace that can only be run by warden
- When first warden is assigned the mute is also invoked
- removed unused imports

Currently there are no config stuff as I haven't look into that. There is an admin bypass for css_peace

### CODE REVIEW

File: mod/Jailbreak.Warden/Global/WardenBehavior.cs [hunk 1]
```diff
@@ -66,7 +70,23 @@ public bool TrySetWarden(CCSPlayerController controller)
 			.ToAllCenter();
 
 		_logs.Append( _logs.Player(_warden), "is now the warden.");
+
+		// makes sure the second person (and people thereafter) who claim warden are not labelled as "first warden"
+		if (_currentWardenIsFirst && _firstWarden)
```
Comment (MSWS) [inline mod/Jailbreak.Warden/Global/WardenBehavior.cs:75]: Confusing variable / if check here.

File: mod/Jailbreak.Warden/Global/WardenPeaceBehaviour.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+namespace Jailbreak.Warden.Global;
+
+public class WardenPeaceBehaviour : IPluginBehavior, IWardenPeaceService
+{
+
+    private readonly IWardenService _wardenService;
+    private readonly ICoroutines _coroutines;
+    private readonly IEventsService _eventsService;
+
+    private readonly float _muteTime = 10.0f;
```
Comment (MSWS) [inline mod/Jailbreak.Warden/Global/WardenPeaceBehaviour.cs:17]: If possible distinguish this variable from https://github.com/edgegamers/Jailbreak/pull/69/files#diff-e81c196d0b0e42fe347b9e3ee80c463e42c9aa74b8baec3127ed11383c01feaeR16

File: public/Jailbreak.Public/Mod/Warden/IWardenPeaceService.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+namespace Jailbreak.Public.Mod.Warden;
+
+public interface IWardenPeaceService
+{
+
+    public CCSPlayerController? GetWarden();
+
+    // todo document saying that by default all admins SHOULD bypass this mute
+    // not implemented bypass yet
+    public void PeaceMute(float time, bool exemptWarden = false);
```
Comment (MSWS) [inline public/Jailbreak.Public/Mod/Warden/IWardenPeaceService.cs:13]: Warden should **always** be exempt.

**Review by MSWS — CHANGES_REQUESTED** [review]

File: mod/Jailbreak.Warden/Commands/WardenPeaceCommandsBehavior.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    {
+
+        if (invoker == null) return;
+
+        CCSPlayerController? warden = _peaceService.GetWarden();
+
+        if (warden == null) return;
+
+        // we only want the warden to be able to run this!
+        if (!invoker.Equals(warden)) return;
```
Comment (MSWS) [inline mod/Jailbreak.Warden/Commands/WardenPeaceCommandsBehavior.cs:36]: Not a fan of wrapping around already written logic for checking if activator is warden. Please use https://github.com/edgegamers/Jailbreak/blob/main/public/Jailbreak.Public/Mod/Warden/IWardenService.cs#L15

File: mod/Jailbreak.Warden/Global/WardenPeaceBehaviour.cs [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+    public CCSPlayerController? GetWarden()
+    {
+        return _wardenService.Warden;
+    }
+
+    public void PeaceMute(float time, bool exemptWarden = false)
+    {
+
+        List<CCSPlayerController> alreadyMuted = new List<CCSPlayerController>();
```
Comment (MSWS) [inline mod/Jailbreak.Warden/Global/WardenPeaceBehaviour.cs:43]: Why keep track of already muted players?

Comment (P250) [reply mod/Jailbreak.Warden/Global/WardenPeaceBehaviour.cs:43]: Good point 😆

File: mod/Jailbreak.Rebel/RebelManager.cs [hunk 6]
```diff
@@ -6,14 +6,13 @@
...
 using Jailbreak.Public.Extensions;
-using Jailbreak.Public.Mod.Logs;
 using Jailbreak.Public.Mod.Rebel;
 
 namespace Jailbreak.Rebel;
 
 public class RebelManager : IPluginBehavior, IRebelService
 {
-    private IRebelNotifications notifs;
+    //private IRebelNotifications notifs;
```
Comment (P250) [inline mod/Jailbreak.Rebel/RebelManager.cs:15]: Unused variable? :)

File: lang/Jailbreak.English/Generic/GenericCommandNotifications.cs [hunk 7]
```diff
@@ -10,7 +9,7 @@ namespace Jailbreak.English.Generic;
 
 public class GenericCommandNotifications : IGenericCommandNotifications, ILanguage<Formatting.Languages.English>
 {
-	public static FormatObject PREFIX =
+	public static FormatObject GENERIC_PREFIX =
```
Comment (P250) [inline lang/Jailbreak.English/Generic/GenericCommandNotifications.cs:12]: Renamed all prefix's so I can statically include them in my Notification implementation

File: lang/Jailbreak.English/Warden/WardenNotifications.cs [hunk 8]
```diff
@@ -11,7 +10,7 @@ namespace Jailbreak.English.Warden;
 
 public class WardenNotifications : IWardenNotifications, ILanguage<Formatting.Languages.English>
 {
-	public static FormatObject PREFIX = new HiddenFormatObject($" {ChatColors.Lime}[{ChatColors.Green}WARDEN{ChatColors.Lime}]")
+	public static readonly FormatObject WARDEN_PREFIX = new HiddenFormatObject($" {ChatColors.Lime}[{ChatColors.Green}WARDEN{ChatColors.Lime}]")
```
Comment (P250) [inline lang/Jailbreak.English/Warden/WardenNotifications.cs:13]: I should make make it readonly for all of them :P

File: mod/Jailbreak.Warden/Global/WardenPeaceBehaviour.cs (hunk 2 above)
Comment (P250) [reply mod/Jailbreak.Warden/Global/WardenPeaceBehaviour.cs]: Done

File: public/Jailbreak.Public/Mod/Warden/IWardenPeaceService.cs (hunk 3 above)
Comment (P250) [reply public/Jailbreak.Public/Mod/Warden/IWardenPeaceService.cs]: Done

File: mod/Jailbreak.Warden/Commands/WardenPeaceCommandsBehavior.cs (hunk 4 above)
Comment (P250) [reply mod/Jailbreak.Warden/Commands/WardenPeaceCommandsBehavior.cs]: Also done

**Comment by P250** [comment]
An important comment is I need to add support for warmup...

File: mod/Jailbreak.Warden/Global/WardenPeaceBehaviour.cs [hunk 9]
```diff
@@ -23,6 +23,8 @@ namespace Jailbreak.Warden.Global;
 /// Any unmute functions also try to respect this.
 /// There are a number of event listeners to handle edge cases, such as removing peace-mute when the round ends, or more general things such as 
 /// enabling peace-mute for Terrorists when the round begins.
+/// If a player is alive they should either be able to speak unless peace-mute is active or an admin-mute is active.
+/// If a player is dead they should never be able to speak to alive players, and their admin-mute status should be respected.
```
Comment (P250) [inline mod/Jailbreak.Warden/Global/WardenPeaceBehaviour.cs:27]: todo clarify if this is true or not

File: mod/Jailbreak.Warden/Global/WardenPeaceBehaviour.cs [hunk 10]
```diff
@@ -308,6 +297,19 @@ public void UnmutePrevMutedPlayers(MuteReason reason, params CsTeam[] targets)
 
     }
 
+    public void UnmutePrevMutedDeadPlayers()
+    {
+        if (_roundEnd)
```
Comment (P250) [inline mod/Jailbreak.Warden/Global/WardenPeaceBehaviour.cs:302]: todo I should remove this check from inside the function and have the method that calls this UnmutePrevMutedDeadPlayers() function check for round end instead.

**Pull request closed without merging**